middlewares/authn.js

```diff
@@ -51,9 +51,9 @@ module.exports = function(req, res, next) {
 
     // otherwise look for auth cookies
     if (req.cookies.pl_authn == null) {
-        logger.error('no authn cookie');
+        //logger.error('no authn cookie');
+        res.cookie('preAuthUrl', req.originalUrl);
         res.redirect('/');
-        return;
```

Comment:
Why can we delete the `return` here? The `res.redirect()` doesn't actually return from this handler. If we are inside this `if {}` we don't want to run the subsequent code, which is why we had the `return`, or am I misunderstanding this?

Reply:
It's probably an oversight/bug that the return was removed. I think it makes it clearer that function is a terminal branch.

But it works because `res.redirect()` runs `res.end()` -- it's not a pass-through res function. https://github.com/expressjs/express/blob/9f019c8c6966736803a65eb4a96d0e7e87e85ede/lib/response.js#L908